src/interfaces/ws.js

```diff
@@ -288,6 +296,16 @@ module.exports = function(app) {
     )
   }
 
+  function processReuestQuery(spark, msg) {
+    queryRequest(msg.requestId)
+      .then(reply => {
+        spark.write(reply)
+      })
+      .catch(err => {
+        console.log(err)
```

Comment:
The client will never hear a response in case of an error, for example with a bogus requestId (a server restart purges all, right?). 

I don't think we thought of error responses in the spec, but sending something helpful back to the client might save somebody some debugging time the future..

Reply:
request/response have `state` and `statusCode` members - it would make sense to use those for that